### PR TITLE
Switch from camera "bounds" to "viewport"

### PR DESCRIPTION
**Pull request by fernandojsg**
As discussed in https://github.com/mrdoob/three.js/issues/16361 and https://github.com/mrdoob/three.js/pull/16316#discussion_r280008571

### CODE REVIEW

File: examples/js/objects/Reflector.js [hunk 1]
```diff
@@ -172,19 +172,9 @@ THREE.Reflector = function ( geometry, options ) {
 
 		// Restore viewport
 
-		var bounds = camera.bounds;
+		if ( camera.viewport !== undefined ) {
```
Comment (mrdoob) [inline examples/js/objects/Reflector.js]: Sorry, one more thing. Can these checks be `if ( camera.isArrayCamera ) {`.

Comment (fernandojsg) [reply examples/js/objects/Reflector.js]: @mrdoob done!

**Comment by mrdoob** [comment]
Thanks!

File: examples/js/objects/Reflector.js [hunk 2]
```diff
@@ -172,19 +172,9 @@ THREE.Reflector = function ( geometry, options ) {
...
-			renderer.getSize( size );
-			var pixelRatio = renderer.getPixelRatio();
-
-			viewport.x = bounds.x * size.width * pixelRatio;
-			viewport.y = bounds.y * size.height * pixelRatio;
-			viewport.z = bounds.z * size.width * pixelRatio;
-			viewport.w = bounds.w * size.height * pixelRatio;
-
-			renderer.state.viewport( viewport );
+			renderer.state.viewport( camera.viewport );
```
Comment (Mugen87) [inline examples/js/objects/Reflector.js]: Um, this code does not look correct. If `camera` is of type `ArrayCamera`, then using `viewport` is invalid. `viewport` is supposed to be a property of sub-cameras and not `ArrayCamera` itself, right?

Comment (Mugen87) [inline examples/js/objects/Reflector.js]: Um, this code does not look correct. If camera is of type `ArrayCamera`, then using `viewport` is invalid. `viewport` is supposed to be a property of sub-cameras and not `ArrayCamera` itself, right?

So maybe like so:
```
var viewport = camera.viewport;

if ( viewport !== undefined ) {

    renderer.state.viewport( camera.viewport );

}
```

Comment (Mugen87) [reply examples/js/objects/Reflector.js]: I've tested this fix today with my Pixel 1 and it seems to solve the broken sandbox example.

**Pull request merged**